libs/blocks/global-navigation/blocks/profile/signIn.js

```diff
@@ -0,0 +1,53 @@
+import { toFragment, getFedsPlaceholderConfig } from '../../utilities/utilities.js';
+import { replaceKey } from '../../../../features/placeholders.js';
+
+const signIn = () => {
+  if (typeof window.adobeIMS?.signIn !== 'function') return;
```

Comment:
So if ims is not loaded yet, a call to this function simply does nothing by returning? What does that mean for the end-user?

Reply:
The logic only gets to here if IMS is ready, as seen [here](https://github.com/adobecom/milo/pull/601/files/10037426414c6f9ff894179cd240d95f19ec3519#diff-628bf5a25903978c5b36da4943ec813da171346d24bd64f80907f012a840b098R165).

When IMS executes the `onReady` callback, we initiate the profile decoration logic. This condition is an extra safeguard in the off chance that `signIn` is not a function.